### PR TITLE
Add snapshot to add-on update dialog.

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Add snapshot to add-on update dialog.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+  public async showDialog(
+    params: SupervisorDialogSupervisorAddonUpdateParams
+  ): Promise<void> {
+    this._opened = true;
+    this.addon = params.addon;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
```
Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:49]: add a `dialog-closed` event

Comment (ludeeus) [reply hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:49]: for what?

Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:49]: Should we reset `_createSnapshot`?

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  }
+
+  private _toggleSnapshot() {
+    this._createSnapshot = !this._createSnapshot;
+  }
+
+  private async _update() {
+    if (this._createSnapshot) {
+      try {
+        this._action = "snapshot";
```
Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:117]: take this out of the try

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+          homeassistant: false,
+        });
+      } catch (err) {
+        this._error = extractApiErrorMessage(err);
+        return;
+      }
+    }
+
+    try {
+      this._action = "update";
```
Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:130]: take this out of the try

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    if (this._createSnapshot) {
+      try {
+        this._action = "snapshot";
+        await createHassioPartialSnapshot(this.hass, {
+          name: `addon_${this.addon.slug}_${this.addon.version}`,
+          addons: [this.addon.slug],
+          homeassistant: false,
+        });
+      } catch (err) {
+        this._error = extractApiErrorMessage(err);
```
Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:124]: `this._action` should be reset?

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        return;
+      }
+    }
+
+    try {
+      this._action = "update";
+      await updateHassioAddon(this.hass, this.addon.slug);
+    } catch (err) {
+      this._error = extractApiErrorMessage(err);
```
Comment (bramkragten) [inline hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:133]: `this._action` should be reset?

File: hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts (hunk 1 above)
Comment (bramkragten) [reply hassio/src/dialogs/addon/dialog-supervisor-addon-update.ts:49]: handling of history back

**Review by bramkragten — APPROVED** [review]

**Pull request merged**